[ROCm] Support all ROCm supported uarchs simultaneously: gfx803, gfx900, gfx906

iotamudelta opened the pull request:
Correct misspelled flag.

Remove dependency on debug flag (HCC_AMDGPU_TARGET)

iotamudelta committed:
Support all ROCm supported uarchs simultaneously: gfx803, gfx900, gfx906.

Correct misspelled flag.

Remove dependency on debug flag (HCC_AMDGPU_TARGET)

iotamudelta commented:
@bddppq this is the follow-up to my comment in #17329 . I think this is more robust and also adds unconditional support for all ROCm uarchs, making user's lifes easier.

bddppq commented on cmake/public/LoadHIP.cmake:
```diff
@@ -94,12 +94,6 @@ ELSE()
 
-IF(NOT DEFINED ENV{HCC_AMDGPU_TARGET})
-  SET(HCC_AMDGPU_TARGET gfx900;gfx906)
-ELSE()
```
We can add gfx803 to the default list, but I would still prefer to keep this user-configurable.

iotamudelta commented:
@bddppq sounds good. Can we use another environment variable though? `PYTORCH_ROCM_ARCHS` or something since the current one doubles as a hcc debug flag and with hip-clang incoming, I'd like to make this truly compiler independent.

bddppq commented:
@iotamudelta Sure renaming it sounds good to me

iotamudelta committed:
Return to user-configurable multi-arch

Rename environment to PYTORCH_ROCM_ARCH to avoid collision with hcc
debug flag.

Still add gfx803 to default architectures.

iotamudelta committed:
correct missing else
